google/cloud/storage/object_metadata.h

```diff
@@ -0,0 +1,75 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_OBJECT_METADATA_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_OBJECT_METADATA_H_
+
+#include "google/cloud/storage/internal/common_metadata.h"
+#include <map>
+
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+/**
+ * Represents a Google Cloud Storage Bucket Metadata object.
+ *
+ * @warning This is an incomplete implementation to validate the design. It does
+ * not support changes to the metadata. It also lacks support for ACLs,
+ * encryption keys, and other features.
+ *
+ * TODO(#537) - complete the implementation.
```

Comment:
This comment block should be changed to reflect that this is Object, not Bucket, metadata. Also, the TODO below (#537) is for competing bucket metadata; I'll let you decide whether to make that issue track completing both, or to make a separate issue for completing object metadata.

Reply:
See above.